test/re/frida/MethodTest.java

```diff
@@ -54,6 +58,131 @@ public void replacementPropagatesExceptions() {
+                "var c2 = Java.use('java.lang.Class');" +
+                "try{" +
+                
+                // hook the original
+                "  var orig = c.invoke;" +
+                "  c.invoke.implementation = function(obj, ...args){ orig(obj, args); };" +
+                
+                // now call it and see what happens
+                "  var cl = c.forName('re.frida.MethodTest');" +
+                "  var method = cl.getMethod('ReturnZero', 'int');" +
+                "  var ret = method.invoke();" +
+                "}catch(e){" + 
+                "  var MethodTest = Java.use('re.frida.MethodTest');" +
+                "  MethodTest.Fail('Method.invoke shat the bed: ' + e);" +
+                "}"
+                );
+        assertNull(failString);
+    }*/
+    
+    /*@Test
+    public void TestNativeLibraryLoading() {
+        loadScript("var c = Java.use('java.lang.System');" +
+                "try{" +
+                "  var orig = c.load;" +
+                "  c.load.implementation = function(s){ orig(s); };" +
+                "  c.load('/system/lib/libc.so')" +
+                "}catch(e){" + 
+                "  var MethodTest = Java.use('re.frida.MethodTest');" +
+                "  MethodTest.Fail('System.load() shat the bed: ' + e);" +
+                "}"
+                );
+        assertNull(failString);
+        
+        loadScript("var c = Java.use('java.lang.Runtime');" +
+                "try{" +
+                "  var orig = c.loadLibrary.overload('java.lang.String');" +
+                "  c.loadLibrary.overload('java.lang.String').implementation = function(s){ orig(s); };" +
+                
+                // now look up the function again and call it
+                "  var now = c.loadLibrary.overload('java.lang.String');" +
+                "  now('/system/lib/libc.so')" +
+                "}catch(e){" + 
+                "  var MethodTest = Java.use('re.frida.MethodTest');" +
+                "  MethodTest.Fail('Runtime.loadLibrary() shat the bed: ' + e);" +
+                "}"
+                );
+        assertNull(failString);
+    }
+    
+    static private String failString = null;
+    static private void Fail( String msg )
+    {
+      failString = msg;
+    }*/
+    
+    @Test
+    public void constructorReturnsCorrectType() {
+        loadScript("var c = Java.use('javax.crypto.spec.SecretKeySpec');" +
+                "try {" +
+                "  var orig = c.$init.overload('[B', 'java.lang.String');" +
+                "  c.$init.overload('[B', 'java.lang.String').implementation = function(a, b){ orig.call(this, a, b); };" +
+                
+                // now look up the function again and call it
+                "  var testConstructor = c.$new( [1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1], 'AES' );" +
+                "} catch (e) {" + 
+                "  send('SecretKeySpec: ' + e);" + 
+                "}" + 
+                "send('ok');"
```

Comment:
The `ok` feels like it belongs inside the `try`, at the very end (that is only reached if everything went well).